zfit/models/physics.py

```diff
@@ -576,6 +576,101 @@ def generalized_gaussexptail_func(x, mu, sigmal, alphal, nl, sigmar, alphar, nr)
+
+    lower, upper = limits._rect_limits_tf
+
+    return gaussexptail_integral_func(mu, sigma, alpha, n, lower, upper)
+
+
+@z.function(wraps="tensor", keepalive=True)
+def gaussexptail_integral_func(mu, sigma, alpha, n, lower, upper):
+    sqrt_pi_over_two = np.sqrt(np.pi / 2)
+    sqrt2 = np.sqrt(2)
+
+    abs_sigma = znp.abs(sigma)
+    abs_alpha = znp.abs(alpha)
+    tmin = (lower - mu) / abs_sigma
+    tmax = (upper - mu) / abs_sigma
+
+    alpha_negative = tf.less(alpha, 0)
+    # do not move on two lines, logic will fail...
+    tmax, tmin = (
+        znp.where(alpha_negative, -tmin, tmax),
+        znp.where(alpha_negative, -tmax, tmin),
+    )
+
+    guass_entire_integral = abs_sigma * sqrt_pi_over_two * (tf.math.erf(tmax / sqrt2) - tf.math.erf(tmin / sqrt2))
+    exp_entire_integral = (
+        abs_sigma / n * znp.exp(-0.5 * znp.square(abs_alpha) + n * abs_alpha) * (znp.exp(n * tmax) - znp.exp(n * tmin))
+    )
+    gauss_integral_alpha_tmax = (
+        abs_sigma * sqrt_pi_over_two * (tf.math.erf(tmax / sqrt2) - tf.math.erf(-abs_alpha / sqrt2))
+    )
+    exp_integral_alpha_tmin = (
+        abs_sigma
+        / n
+        * znp.exp(-0.5 * znp.square(abs_alpha) + n * abs_alpha)
+        * (znp.exp(-n * abs_alpha) - znp.exp(n * tmin))
+    )
+    integral_sum = exp_integral_alpha_tmin + gauss_integral_alpha_tmax
+
+    conditional_integral = tf.where(tf.less_equal(tmax, -abs_alpha), exp_entire_integral, integral_sum)
+    result = tf.where(tf.greater_equal(tmin, -abs_alpha), guass_entire_integral, conditional_integral)
+    if result.shape.rank != 0:
+        result = tf.gather(result, 0, axis=-1)
+    return result
+
+
+def generalized_gaussexptail_integral(limits, params, model):
+    del model
+    mu = params["mu"]
+    sigmal = params["sigmal"]
+    alphal = params["alphal"]
+    nl = params["nl"]
+    sigmar = params["sigmar"]
+    alphar = params["alphar"]
+    nr = params["nr"]
+
+    lower, upper = limits._rect_limits_tf
+    lower = lower[:, 0]
+    upper = upper[:, 0]
+
+    return generalized_gaussexptail_integral_func(
+        mu=mu,
+        sigmal=sigmal,
+        alphal=alphal,
+        nl=nl,
+        sigmar=sigmar,
+        alphar=alphar,
+        nr=nr,
+        lower=lower,
+        upper=upper,
+    )
+
+
+@z.function(wraps="tensor", keepalive=True)
+def generalized_gaussexptail_integral_func(mu, sigmal, alphal, nl, sigmar, alphar, nr, lower, upper):
+    upper_of_lowerint = znp.minimum(mu, upper)
+    integral_left = gaussexptail_integral_func(
+        mu=mu, sigma=sigmal, alpha=alphal, n=nl, lower=lower, upper=upper_of_lowerint
+    )
+    left = tf.where(tf.less(mu, lower), znp.zeros_like(integral_left), integral_left)
+
+    lower_of_upperint = znp.maximum(mu, lower)
+    integral_right = gaussexptail_integral_func(
+        mu=mu, sigma=sigmar, alpha=-alphar, n=nr, lower=lower_of_upperint, upper=upper
+    )
+    right = tf.where(tf.greater(mu, upper), znp.zeros_like(integral_right), integral_right)
```

Comment:
why exactly is this split needed and we can't do `tf.where(bool, left, right)`? Is it maybe with NaNs and grad?

Reply:
I just did the same thing that was done for the crystal ball case. Will check.

Reply:
This just says that if `mu` is greater than the upper limit, set the right integral to zero because it's outside our integration range. The equaivalent exists for the left integral as well and then you sum the two. Can you show me what you think can be removed here?

Reply:
ah yes, I misread! All fine